src/legacy/ui/public/url/url.js

```diff
@@ -47,6 +50,31 @@ export function KbnUrlProvider($injector, $location, $rootScope, $parse, Private
    */
   const self = this;
 
+  /**
+   * Override the behavior of angular's $location.url() to encode/decode
+   * URL parameters
+   * @param {String} url
+   * @returns $location
+   */
+  $location.url = url => {
+    if (typeof url === 'undefined') {
+      return $location.$$url;
+    }
+
+    const match = PATH_MATCH.exec(url);
+
+    if (match[1] || url === '') {
+      $location.path(($location.$$html5 ? decodeURI : decodeURIComponent)(match[1]));
+    }
+    if (match[2] || match[1] || url === '') {
+      $location.search(match[3] || '');
+    }
+
+    $location.hash(match[5] || '');
+
+    return $location;
+  };
```

Comment:
This is awesome! Good job!
The reason Angular didn't fix this (they actually did but had to revert) was because too much of the ecosystem was relying in the broken behaviour. We should make sure that's not the case in Kibana.

Reply:
Thank you! Just a note: I walked through every usage of parameter encoding in Kibana I could find to see if there was any codec work around the URL and these were all I could find.

Reply:
>  I walked through every usage of parameter encoding in Kibana I could find to see if there was any codec work around the URL and these were all I could find.
Wow, that sounds painful 😁 

I'll spin this up tomorrow, and let you know how APM fares.

Reply:
Hmm.. just tried it and it seems to break the url. 
This url:
```
http://localhost:5601/app/apm#/opbeans-node/transactions/request/GET%20%2Fapi%2Fcustomers
```

Redirects to:
```
http://localhost:5601/app/apm#/opbeans-node/transactions/request/GET%20/api/customers
```

(Notice how the encoded forward slashes gets decoded)